mvpoly/src/prime.rs

```diff
@@ -542,4 +549,138 @@ impl<F: PrimeField, const N: usize, const D: usize> From<F> for Dense<F, N, D> {
     }
 }
 
+impl<F: PrimeField, const N: usize, const D: usize> From<ConstantExprInner<F>> for Dense<F, N, D> {
+    fn from(expr: ConstantExprInner<F>) -> Self {
+        match expr {
+            // The unimplemented methods might be implemented in the future if
+            // we move to the challenge into the type of the constant
+            // terms/expressions
+            // Unrolling for visibility
+            ConstantExprInner::Challenge(ChallengeTerm::Alpha) => {
+                unimplemented!("The challenge alpha is not supposed to be used in this context")
+            }
+            ConstantExprInner::Challenge(ChallengeTerm::Beta) => {
+                unimplemented!("The challenge beta is not supposed to be used in this context")
+            }
+            ConstantExprInner::Challenge(ChallengeTerm::Gamma) => {
+                unimplemented!("The challenge gamma is not supposed to be used in this context")
+            }
+            ConstantExprInner::Challenge(ChallengeTerm::JointCombiner) => {
+                unimplemented!(
+                    "The challenge joint combiner is not supposed to be used in this context"
+                )
+            }
+            ConstantExprInner::Constant(ConstantTerm::EndoCoefficient) => {
+                unimplemented!(
+                    "The constant EndoCoefficient is not supposed to be used in this context"
+                )
+            }
+            ConstantExprInner::Constant(ConstantTerm::Mds {
+                row: _row,
+                col: _col,
+            }) => {
+                unimplemented!("The constant Mds is not supposed to be used in this context")
+            }
+            ConstantExprInner::Constant(ConstantTerm::Literal(c)) => Dense::from(c),
+        }
+    }
+}
+
+impl<F: PrimeField, const N: usize, const D: usize> From<Operations<ConstantExprInner<F>>>
+    for Dense<F, N, D>
+{
+    fn from(op: Operations<ConstantExprInner<F>>) -> Self {
+        use kimchi::circuits::expr::Operations::*;
+        match op {
+            Atom(op_const) => Self::from(op_const),
+            Add(c1, c2) => Self::from(*c1) + Self::from(*c2),
+            Sub(c1, c2) => Self::from(*c1) - Self::from(*c2),
+            Mul(c1, c2) => Self::from(*c1) + Self::from(*c2),
```

Comment:
This should use mul, I would imagine (not +).

Reply:
I'm a little worried about the effectiveness of the tests for not catching this, as well, but that's another issue.

Reply:
I feel like I don't have the necessary background knowledge to properly review this commit.

cc @marcbeunardeau88

Reply:
huhu, nice catch :see_no_evil:

Reply:
![image](https://github.com/user-attachments/assets/e8dfb019-4768-47ae-8aa4-ea817e98073c)